pkg/operation/cloudbotanist/alicloudbotanist/infrastructure.go

```diff
@@ -227,6 +227,9 @@ func cleanSnapshots(bucketName, storageEndpoint, accessKeyID, accessKeySecret st
 		for _, object := range lsRes.Objects {
 			snapshots = append(snapshots, object.Key)
 		}
+		if len(snapshots) == 0 {
```

Comment:
I think we might miss some case:

What if there are more responses via `bucket.ListObjects()` in the next loop whose buckets are not empty?

Reply:
Hi Tim, thanks for your comments. `cleanSnapshots` only handles one bucket once call, the loop inside traverses and returns all objects via `bucket.ListObjects()`. There does exist more responses case, but I suppose once `bucket.ListObjects()` returns empty, it means no other objects existing in the bucket, so it would be safe to return.
Of course if I change the code like this, suppose it would be safer:
```
if len(snapshots) > 0 {
  if _, err = bucket.DeleteObjects(snapshots); err != nil {
    return err
  }
}
```

Reply:
Okay, thanks for the explanation.

I really like your suggestion, though. Can you change the PR accordingly?

Reply:
Sure. Just changed.